### ISSUE TITLE
v3.0 of GB Studio doesn't work?

### ISSUE THREAD

**Issue by kenyonbowers** [asiekierka/gbsenpai#2]
Hello, I've seen that v3.0 of GB Studio does not work... Don't know what else to say...

**Comment by asiekierka** [comment]
That is correct. As per the README: `gbsenpai should be compatible with source exports made in GB Studio v2.0.0-beta5. Other versions are currently not supported.`

**Comment by kenyonbowers** [comment]
Will you support v3.0 in the future?

**Comment by asiekierka** [comment]
I evaluated the necessary effort. It's not impossible, but I do not currently have the resources to develop a solution on volunteer time.

**Comment by kenyonbowers** [comment]
What would it take to make it v3.0 compatible?

**Comment by asiekierka** [comment]
The very rough version is: writing a converter from GBDK object files to something I can work with, re-applying gbsenpai patches to the new v3 engine, and ideally moving away from the dodgy Makefile-based solution into something that is easier to use by end users.

**Comment by kenyonbowers** [comment]
What converter?

**Comment by kenyonbowers** [comment]
Why not just change the files GBSenpai looks for to the equivalent?

**Comment by asiekierka** [comment]
v3.0 uses SDCC assembly files with macros which probably won't work with GCC's assembler. So I need to compile them with SDCC, then convert them to something a modern linker can understand.

**Comment by kenyonbowers** [comment]
Oh

**Comment by maxoakland** [comment]
I know it would be a lot of work but it would be so amazing.

**Comment by dxkill** [comment]
yea pls do it

**Comment by neoskedar** [comment]
> I evaluated the necessary effort. It's not impossible, but I do not currently have the resources to develop a solution on volunteer time.

@asiekierka  What would it cost for you to develop this?  I'd be willing to pay some and I'm sure others would too!
